packages/eslint-plugin/typings/typescript.d.ts

```diff
@@ -25,6 +25,15 @@ declare module 'typescript' {
   }
 
   interface Type {
+    /**
+     * Private, unique identifier for the type as cached by the TypeScript program.
+     *
+     * @remarks
+     * This should rarely be used, and then only to check for reference equality
+     * between types (since we don't have an available isAssignableTo API).
+     */
+    id?: string;
```

Comment:
I'm not too pleased with relying on this `id` to exist... but short of a type relationship API in TypeScript (https://github.com/microsoft/TypeScript/issues/9879), I don't see a better way...?

Reply:
unless I'm mistaken - I think that you should be able to use `===` to compare.

Reply:
Ha! I could have sworn I'd tried that and seen evidence to the contrary. No longer. Excellent. Thank you 😄.